src/boot/efi/SECURITY.md

```diff
@@ -0,0 +1,92 @@
+# UEFI Components Security Posture
+The systemd project provides a UEFI boot menu, `systemd-boot`, and a stub that can wrap a Linux kernel in a
+PE binary, adding various features, `systemd-stub`. These components fully support UEFI SecureBoot, and
+this document will describe their security posture and how they comply with industry-standard expectations
+for UEFI SecureBoot workflows.
+
+## Fundamental Security Design Goals
+The fundamental security design goals for these components are separation of security policy logic from the
+rest of the functionality, achieved by offloading security-critical tasks to the firmware or earlier stages
+of the boot process (e.g.: `Shim`).
+
+When SecureBoot is enabled, these components are designed to avoid executing, loading or using
+unauthenticated payloads that could compromise the boot process, with special care taken for anything that
+could affect the system before `ExitBootServices()` has been called. For example, when additional resources
+are loaded, if running with SecureBoot enabled, they will be validated before use. The only exceptions are
+the bootloader's own textual configuration files, and parsing metadata out of images for displaying purposes
+only. There are no build time or runtime configuration options that can be set to weaken the security model
+of these components when SecureBoot is enabled.
```

Comment:
so, in sd-stub when we pass control to the inner kernel we actually disable authentication first for that single invocation, via code that some people might consider a "vaguely horrible hack", i.e. see install_security_override(). (by the standards of what shim is doing it's probably in the category of "beautiful exemplary code" though)

it might be worth mentioning this explicitly, since people might otherwise be confused why this is safe.

i.e. mention somewhere: "systemd-stub will disable SecureBoot authentication for the payload kernel it wraps, in order to avoid redundant duplicative authentication of the image, given that that the payload kernel is already authenticated because it is part of the systemd-stub EFI binary which already was SecureBoot authenticated at this point".